plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergParquetComplexTypePredicatePushDown.java

```diff
@@ -0,0 +1,76 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.Session;
+import io.trino.testing.BaseTestFileFormatComplexTypesPredicatePushDown;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestIcebergParquetComplexTypePredicatePushDown
+        extends BaseTestFileFormatComplexTypesPredicatePushDown
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return IcebergQueryRunner.builder()
+                .addIcebergProperty("iceberg.file-format", "PARQUET")
+                .build();
+    }
+
+    // limits the number of rows per row group size. The total row count remains the same, this introduces more row groups.
+    @Override
+    protected final Session getSession()
+    {
+        return Session.builder(super.getSession())
+                .setCatalogSessionProperty("iceberg", "parquet_writer_page_size", "100B")
+                .setCatalogSessionProperty("iceberg", "parquet_writer_block_size", "100B")
+                .setCatalogSessionProperty("iceberg", "parquet_writer_batch_size", "10")
+                .build();
+    }
+
+    // The Iceberg table scan differs from Hive in that the coordinator also uses file statistics when generating the splits .
+    // As a result, if the predicates fall outside the bounds of the file statistics,
+    // the split is not created for the worker and worker won't call getParquetTupleDomain().
+    // The test increased the number of row groups and introduced predicates that are within the file statistics but outside the statistics of the row groups.
+    @Test
+    public void testIcebergParquetRowTypeRowGroupPruning()
+    {
+        String tableName = "test_nested_column_pruning_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (col1Row ROW(a BIGINT, b BIGINT))");
+        assertUpdate("INSERT INTO " + tableName + " SELECT * FROM unnest(transform(SEQUENCE(1, 10000), x -> ROW(ROW(x*2, 100))))", 10000);
+
+        // col1Row.a only contains even numbers, in the range of [2, 20000].
+        // The test has 10 rows per row group, [2, 20], [22, 40], ... [19984, 20000]
+        // 21 is a value between [2, 20000] but is an odd number, so won't be discarded by Iceberg table's statistics.
+        // At the meantime, 21 is not within the bound of any row group. So can be discarded by Parquet's row group statistics.
+        assertNoDataRead("SELECT * FROM " + tableName + " WHERE col1Row.a = 21");
+        assertNoDataRead("SELECT * FROM " + tableName + " WHERE col1Row.a IS NULL");
```

Comment:
`assertNoDataRead` is currently a bit misleading because it builds on `processedInputDataSize` and not `physicalInputDataSize` 

However this change relieves the engine of dealing with additional computations when the data leaves the parquet reader. Very good catch @leetcode-1533